### ISSUE TITLE
Ajaxprogressive load and ajax filtering 

### ISSUE THREAD

**Issue by inspectorman** [olifolkerd/tabulator#1271]
Hello dear,

First of all, thanks for this amazing library. I'm using the tabulator with ajax data being loaded. I have configured the ajaxFiltering and ajaxProgressiveLoad on scroll feature. According to docs, the data passed when ajaxprogressiveLoad should be:
`{
    "last_page":15, //the total number of available pages (this value must be greater than 0)
    "data":[ // an array of row data objects
        {id:1, name:"bob", age:"23"}, //example row data object
    ]
}`
However, it works without filters, when ajaxFiltering is on, it seems to expect data in the format:
`[
    {id:1, name:"Billy Bob", age:"12", gender:"male", height:1, col:"red", dob:"", cheese:1},
    {id:2, name:"Mary May", age:"1", gender:"female", height:2, col:"blue", dob:"14/05/1982", cheese:true},
]`
However, at this point, the progressive load won't work.
So I got the error "Unable to process data due to invalid data type " when filtering but not on loading the initial data.

My tabulator config is:
```js$("#tabulator-table").tabulator({
    height:400, // set height of table (in CSS or here), this enables the Virtual DOM and improves render speed dramatically (can be any valid css height value)
    layout:"fitColumns", //fit columns to width of table (optional)
    ajaxURL:'<?= $this->Url->build(["action" => "getInspectionPlanAjax","controller" => "reports"]); ?>',
    ajaxProgressiveLoad:"scroll",
	paginationSize:30,
	//pagination:"remote",
	ajaxSorting:false,
	ajaxFiltering:true,
	placeholder:"Cargando...",
	columns:[ //Define Table Columns
		{title:'Departamento', field:'cml.circuit.corrosion_loop.unit.department.name', width:150,headerFilter:true},
		{title:'Unidad', field:'cml.circuit.corrosion_loop.unit.name', width:150,headerFilter:true},
		{title:'Lazo', field:'cml.circuit.corrosion_loop.name', width:150,headerFilter:true},
		{title:'Circuit', field:'cml.circuit.name', width:150,headerFilter:true},
		{title:'Group', field:'cml.group.name', width:150,headerFilter:true},
		{title:'Cml', field:'cml.number', width:50},
		{title:'Tml', field:'number', width:50},
		{title:'Tipo', field:'cml.cml_type.name', width:50},
		{title:'Nom Diam', field:'nom_diam', width:50},
		{title:'Nom Thk', field:'nom_thk', width:50},
		{title:'Material', field:'material.name', width:150},
		{title:'Used Cr', field:'used_cr', width:50},
		{title:'Why Cr', field:'why_cr', width:50},
		{title:'Ret Thk', field:'ret_thk', width:50},
		{title:'Car', field:'car', width:50},
		{title:'Next I Date', field:'next_i_date', width:150,headerFilter:true},
		{title:'Rem Life Date', field:'rem_life_date', width:150,headerFilter:true},
    ],
	paginationDataSent:{
		"size":"perPage",
        "filter":"filter",
        "filter_value":"filter_value",
        "filter_type":"filter_type",
    },
    rowClick:function(e, row){ //trigger an alert message when the row is clicked
        alert("Row " + row.getData().id + " Clicked!!!!");
    },
});
$("#tabulator-table").tabulator();
```
thanks in advance for your time,

Javier

**Comment by inspectorman** [comment]
Hello dear Oli,

I've made a little workaround to solve my issue, I've modified the RowManager.prototype._genRemoteRequest in tabulator.js file , exactly on line 3186, making data to be the index "data" in the ajax response.

That way it's working now. 
Hopefully there would be a better approach, but by the moment it solves my issue.

`RowManager.prototype._genRemoteRequest = function () {

      var self = this,
          table = self.table,
          options = table.options,
          params = {};
	var data2;
      if (table.extExists("page")) {

        //set sort data if defined

        if (options.ajaxSorting) {

          var sorters = self.table.extensions.sort.getSort();

          sorters.forEach(function (item) {

            delete item.column;
          });

          params[self.table.extensions.page.paginationDataSentNames.sorters] = sorters;
        }

        //set filter data if defined

        if (options.ajaxFiltering) {

          var filters = self.table.extensions.filter.getFilters(true, true);

          params[self.table.extensions.page.paginationDataSentNames.filters] = filters;
        }

        self.table.extensions.ajax.setParams(params, true);
      }

      table.extensions.ajax.sendRequest(function (data) {
		if (self.table.extensions.page.progressiveLoad) {
			data2 = data[self.table.extensions.page.paginationDataReceivedNames.data];
		}else {
			data2 =data;
		}
        self.setData(data2);
      });
    };`

**Comment by olifolkerd** [comment]
Hey @inspectorman 

Thanks for getting in touch,

i will have a look into this next weekend, could you confirm which version of Tabulator you are using?

Cheers

Oli :)

**Comment by adnanganie** [comment]
How i filter data in whole table

**Comment by inspectorman** [comment]
@olifolkerd The tabulator Version is: Tabulator v3.5.3

Regards,

**Comment by olifolkerd** [comment]
Hey All,

I have just pushed a fix for this to the 4.0 branch and will include it in the next release.

Cheers

Oli
